source/developers/file-transfer-protocol.rst

```diff
@@ -0,0 +1,226 @@
+
+- ``get``
+- ``put``
+- ``ls``
+- ``mkdir``
+- ``rmdir``
+- ``delete``
+
+MinIO does not support either ``append`` or ``rename`` operations.
+
+Considerations
+--------------
+
+Versioning
+~~~~~~~~~~
+
+You cannot use ``FTP`` to read specific :ref:`object versions <minio-bucket-versioning>` other than the latest version.
+
+- For read operations, MinIO only returns the latest version of the requested object(s) to the ftp client.
+- For write operations, MinIO applies normal versioning behavior for matching object names.
+
+Use an S3 API Client, such as the :ref:`MinIO Client <minio-client>`.
+
+Authentication and Access
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``FTP`` access requires the same authentication as any other S3 client.
+MinIO supports the following authentication providers:
+
+- :ref:`MinIO IDP <minio-internal-idp>` users and their service accounts
+- :ref:`Active Directory/LDAP <minio-external-identity-management-ad-ldap>` users and their service accounts
+- :ref:`OpenID/OIDC <minio-external-identity-management-openid>` service accounts
+
+:ref:`STS <minio-security-token-service>` credentials **cannot** access buckets or objects over FTP.
+To use STS credentials to authenticate, you must use an S3 API client or port.
+
+Authenticated users can access buckets and objects based on the :ref:`policies <minio-policy>` assigned to the user or parent user account.
+
+The FTP protocol does not require any of the ``admin:*`` `permissions <minio-policy-mc-admin-actions>`.
+The FTP protocols do not support any of the MinIO admin actions.
+
+Prerequisites
+-------------
+
+- MinIO RELEASE.2023-04-20T17-56-55Z or later.
+- Enable an FTP or SFTP port for the server.
+- A port to use for the FTP commands and a range of ports to allow the FTP server to request to use for the data transfer.
+
+Procedure
+---------
+
+1. Start MinIO with an FTP and/or SFTP port enabled.
+
+   .. code-block:: shell
+      :class: copyable
+
+      minio server http://server{1...4}/disk{1...4}        \
+      --ftp="address=:8021"                                \
+      --ftp="passive-port-range=30000-40000"               \
+      --sftp="address=:8022"                               \
+      --sftp="ssh-private-key=/home/miniouser/.ssh/id_rsa" \
+      ...
+    
+   See the :mc-cmd:`minio server --ftp` and :mc-cmd:`minio server --sftp` for details on using these flags to start the MinIO service.
+   To connect to the an ftp port with TLS (``FTPS``), pass the ``tls-private-key`` and ``tls-public-cert`` keys and values, as well, unless using the MinIO default TLS keys.
+
+   The output of the command should returns a response that includes something like the following:
+
+   .. code-block:: shell
+
+      MinIO FTP Server listening on :8021
+      MinIO SFTP Server listening on :8022
+
+2. Use your preferred ftp client to connect to the MinIO deployment with the user whose policies you want to use to access buckets and objects.
```

Comment:
An idea to break up this long sentence: `Use your preferred ftp client to connect to the MinIO deployment. You must connect as a user whose policies allow access to the desired buckets and objects.`

Reply:
I like it. Thanks.